### PR TITLE
added abort functionality and tests

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: boxsdk/util/chunked_uploader.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def __init__(self, upload_session, content_stream, file_size):
         self._part_array = []
         self._sha1 = hashlib.sha1()
         self._inflight_part = None
+        self._is_aborted = False
```
Comment (Jeff-Meadows) [inline boxsdk/util/chunked_uploader.py]: why is this needed?

File: boxsdk/util/chunked_uploader.py [hunk 2]
```diff
@@ -46,6 +47,21 @@ def start(self):
...
+        :returns:
+            A boolean indication success of the upload abort.
+        :rtype:
+            `bool`
+        """
+        self._content_stream = None
+        self._part_array = []
+        self._inflight_part = None
+        self._is_aborted = True
+        return self._upload_session.abort()
```
Comment (Jeff-Meadows) [inline boxsdk/util/chunked_uploader.py]: I think this is the only line you need for this method. The rest are handled by GC
